qiskit/circuit/parameterexpression.py

```diff
@@ -490,6 +499,52 @@ def __eq__(self, other):
             return len(self.parameters) == 0 and complex(self._symbol_expr) == other
         return False
 
+    def __lt__(self, other):
+        """Check if this parameter expression is less than another parameter expression
+           or a fixed value (only if this is a bound expression).
+        Args:
+            other (ParameterExpression or a number):
+                Parameter expression or numeric constant used for comparison
+        Returns:
+            bool: result of the comparison
+        """
+        if isinstance(other, ParameterExpression):
+            if HAS_SYMENGINE:
+                from sympy import sympify
+
+                return sympify(self._symbol_expr).__lt__(sympify(other._symbol_expr))
+            else:
+                return self._symbol_expr.__lt__(other._symbol_expr)
+        elif isinstance(other, numbers.Number):
+            return len(self.parameters) == 0 and float(self._symbol_expr) < other
```

Comment:
If `len(self.parameters) !=0` this returns `False`, but it should raise an exception.

Reply:
I think the `__eq__` method should also have this behaviour. What do you think?